huaweicloud/services/eg/data_source_huaweicloud_eg_event_streams.go

```diff
@@ -0,0 +1,407 @@
+package eg
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-uuid"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+// @API EG GET /v1/{project_id}/eventstreamings
+func DataSourceEventStreams() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceEventStreamsRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "The region in which to obtain the EG event streams resource.",
+			},
+			"event_streams": {
+				Type:        schema.TypeList,
+				Computed:    true,
+				Description: "The list of event streams.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The ID of the event stream.",
+						},
+						"name": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The name of the event stream.",
+						},
+						"status": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The status of the event stream.",
+						},
+						"description": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The description of the event stream.",
+						},
+						"source": {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "The event source configuration.",
+							Elem:        DataSourceEventStreamsSource(),
+						},
+						"sink": {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "The event sink configuration.",
+							Elem:        DataSourceEventStreamSink(),
+						},
+						"rule_config": {
+							Type:        schema.TypeList,
+							Computed:    true,
+							Description: "The configuration of event rules.",
+							Elem:        DataSourceEventRuleConfig(),
```

Comment:
```suggestion
							Elem:        dataSourceEventSteamRuleConfig(),
```
`Stream` word is missing. other places ditto.

Reply:
done